Week 5. Tutorial | # Taxonomy, Taxa Barplots, Filtering, Phylogenetic Trees

1. Log onto Alpine On-Demand: [ondemand-rmacc.rc.colorado.edu](http://ondemand-rmacc.rc.colorado.edu/ "(opens in a new window)")

2. Navigate to your decomp tutorial directory

3. Open a terminal

4. Go to the taxonomy folder:  
```
cd /scratch/alpine/$USER/decomp_tutorial/taxonomy
```

5.  Load the class node:
```
sinteractive --reservation=aneq505 --time=02:00:00 --partition=amilan --nodes=1 --ntasks=6 --qos=normal
```

6. Load your Qiime2 environment:
```
module purge  
module load qiime2/2024.10_amplicon
```

7. wget the classifier:
```
wget --no-check-certificate https://ftp.microbio.me/greengenes_release/2024.09/2024.09.backbone.v4.nb.qza
```

8. Use our representative reads to classify taxonomy, which will give us our taxonomy.qza output. This will take ~2 minutes.
```
qiime feature-classifier classify-sklearn \--i-reads ../dada2/seqs.qza \--i-classifier 2024.09.backbone.v4.nb.qza \--o-classification taxonomy_gg2.qza
```

9. Let's make our taxonomy into a visualization, transfer to our local computers, and look at it in QIIME2 View.
```
qiime metadata tabulate \--m-input-file taxonomy_gg2.qza \--o-visualization taxonomy_gg2.qzv
```

10. Since there are over 400 samples, we are first going to group by type_days - a metadata column that includes both sample type and days since placement. This will make our results a lot more interpretable
```
qiime feature-table group \--i-table ../dada2/table.qza \--m-metadata-file ../metadata/metadata.txt \--m-metadata-column type_days \--p-mode mean-ceiling \--p-axis sample \--o-grouped-table ../dada2/table_type_days.qza
```

11. Now, using our grouped feature table, we will remove contaminating features like chloroplasts and mitochondria. 
- Things we often want to remove are plant-derived DNA or eukaryotically-derived DNA, but you can remove other contaminates too here by adding that taxonomy to the --p-exclude line. 
- The reason these show up in our data is due to the endosymbiotic theory, that mitochondria & chloroplasts, likely originating as bacteria, became symbionts in cytoplasm of eukaryotic cells. 
- Note that sp004296775 is another chloroplast, it MUST also be removed. see this forum post.Links to an external site.
- ```
  qiime taxa filter-table \--i-table ../dada2/table_type_days.qza \--i-taxonomy taxonomy_gg2.qza \--p-exclude mitochondria,chloroplast,sp004296775 \--p-include c__ \--o-filtered-table ../dada2/table_type_days_nomitochloro.qza
  ```



Taxa Barplots
Now that we have ASVs classified taxonomically, let's generate a taxa barplot to visualize it, transfer it to our local computers, and look at it in QIIME2 View.

1. Move to the taxa plots directory
```
cd ../taxaplots
```

2. Generate the taxa bar plot 
```
qiime taxa barplot \--i-table ../dada2/table_type_days_nomitochloro.qza \--i-taxonomy ../taxonomy/taxonomy_gg2.qza \--o-visualization taxa_barplot_type_days_nomitochloro.qzv
```

3. Check the controls

First, you will need to filter down to just the controls:
```
#First filter samples

qiime feature-table filter-samples \--i-table ../dada2/table.qza \--m-metadata-file ../metadata/metadata.txt \--p-where "[sample_type]='control'" \--o-filtered-table ../dada2/table_controls.qza
```

 Create a taxa barplot with our table with just controls:
```
qiime taxa barplot \--i-table ../dada2/table_controls.qza \--i-taxonomy ../taxonomy/taxonomy_gg2.qza \--m-metadata-file ../metadata/metadata.txt \--o-visualization taxa_barplot_controls.qzv
```

```
qiime taxa filter-table \--i-table ../dada2/table.qza \--i-taxonomy taxonomy_gg2.qza \--p-exclude mitochondria,chloroplast,sp004296775 \--p-include c__ \--o-filtered-table ../dada2/table_nomitochloro.qza  
  
#visualize:   
qiime taxa barplot \--i-table ../dada2/table_nomitochloro.qza \--i-taxonomy ../taxonomy/taxonomy_gg2.qza \--m-metadata-file ../metadata/metadata.txt \--o-visualization taxa_barplot_all_samples.qzv
```


Phylogenetic Trees

1. Go to the tree directory
```
cd ../tree
```

2. Get the reference backbone
```
wget https://ftp.microbio.me/greengenes_release/2022.10/2022.10.backbone.sepp-reference.qza
```

3. 